### DISCUSSION TITLE
Changes to Symbology for webRAVE Rasters & Vectors

### DISCUSSION THREAD

**Discussion by MattReimer** [Riverscapes/RiverscapesXML#117] [RAVE: Symbology/BusinessLogic]
Just a heads-up for @joewheaton @shelbysawyer @lauren-herbine: I've had to make the webRAVE vector symbology JSON a little more complicated. 

## Does it work?

I think so. Take a look a the much better legends:

* [Riverscapes Project (try the DEM and HAND layers)](https://northarrowresearchlabs.github.io/webrave-staging/#/Anabranch/87665ff3-f134-4e22-9020-623ed8ae2b9a)
* [Confinement Project](https://northarrowresearchlabs.github.io/webrave-staging/#/Anabranch/8b5e6234-7635-4b3b-8ac5-b24069bf6768)

## Why do this?

The mapbox export business was great but it didn't have enough information to build any kind of sensible legend in webRAVE.

Apologies for this. I'm trying really hard not to pull the rug out from under everyone as we build symbologies but this was unnavoidable.

Nothing we've done so far is lost. Every JSON file just needs to be augmented to have a little more information.

### Raster Symbology

First some good news: Raster symbology is now WAY easier. Just click a button in QGIS, rename the file it gives you and push to github.

**[Full instructions regarding Raster symbologies for webRAVE](https://github.com/Riverscapes/RiverscapesXML/blob/master/docs/Symbology/webRAVERasters.md)**

### Vector Symbology

I need more information than mapbox can provide in its export. As a result I've had to implement a separate `legend` table object. 

The bare bones of the new JSON file looks like this.

```
{
  "$schema": "https://xml.riverscapes.xyz/Symbology/webRAVEVector.schema.json",
  "legend": [],  <-- this is the new legend table
  "layerStyles": [] <-- this is the mapbox copy and paste
}
```

The [Full instructions on how to build the new JSON files for vectors here](https://github.com/Riverscapes/RiverscapesXML/blob/master/docs/Symbology/webRAVEVectors.md). I've also implemented a schema so that VSCode should catch problems as you type AND I've implented automated validation in Github so you can't actually push anything that doesn't validate without alarm bells going off.

## What does this mean?

I've already implemented everything on the master branch for you.  When you're ready to merge your web symbology pull requests I'll meet with you if you'd like and help you augment them with the legend table.

--------------

Thoughts? Questions?

**Comment by joewheaton** [comment]
@wally-mac, @jtgilbert  and @Cashe93, just a heads up that @MattReimer has been busy working on Warehouse 3.x and so many of the aspirational things we discussed are now realities. We will need to keep a focus on updating our Technical Documentation on how to do the next round of curation. @MattReimer will likely give us much of the MD in discussions and posts like this, but we will need to translate that to some documentation that makes sense to us on how to do the curation. Stay tuned.
